testsuites/syncgateway/functional/tests/test_seq.py

```diff
@@ -100,3 +101,90 @@ def test_seq(params_from_base_test_setup, sg_conf_name, num_users, num_docs, num
     all_doc_caches = [user.cache for user in users]
     all_docs = {k: v for cache in all_doc_caches for k, v in list(cache.items())}
     verify_changes(users, expected_num_docs=num_users * num_docs, expected_num_revisions=num_revisions, expected_docs=all_docs)
+
+
+@pytest.mark.syncgateway
+@pytest.mark.prometheus
+def test_promethus_public_ports(params_from_base_test_setup):
+    """
+    @summary:
+    1. set up prometheus
+    2. Access some of the public api with prometheus port
+    3. Verify it is not accessible
+    """
+    prometheus_enabled = params_from_base_test_setup["prometheus_enabled"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+
+    # Skip the test if prometheus not enabled and sgw version is not
+    if not(prometheus_enabled and sync_gateway_version >= "2.8.2"):
+        pytest.skip('this is prometheus specific test')
+
+    session = Session()
+    sg_db = "db"
+    # 1. set up prometheus
+    prometheus_url = sg_admin_url.replace("4985", "4986")
+
+    # 2. Access some of the public api with prometheus port
+    req = session.get("{}/{}".format(prometheus_url, sg_db))
+    # 3. Verify it is not accessible
+    assert req.status_code == 404, "Expected 404 status, actual {}".format(req.status_code)
+
+    req = session.get("{}/{}/_config".format(prometheus_url, sg_db))
```

Comment:
what about 4986? on SGW config , this port is dedicated to 4986 for prometheus. I could acces 4986 if I enable prometheus,otherwise I cannot. When I run this test with 2.8.2 I got 200 which means I could access public API , fix is public APIs not accessible  and I should get 404 . I got expected with 2.8.3 and failing on 2.8.2. In SGw config, we are not configuring 9090 port, how does sgw knows about the port. Looks like the fix is for 4986 only

Reply:
Ok I got feature, you are testings the stats api not the Prometheus.
Change the variable names to stats. Avoid Promotheus name in the test.

Reply:
my test is updated. We still need to update the framework changes from prometheus to metrics. Let's have action item , otherwise it is confusing in the framework